datafusion/physical-plan/src/joins/hash_join.rs

```diff
@@ -665,6 +668,8 @@ impl ExecutionPlan for HashJoinExec {
             reservation,
             state: HashJoinStreamState::WaitBuildSide,
             build_side: BuildSide::Initial(BuildSideInitialState { left_fut }),
+            batch_size,
```

Comment:
I am not comfortable selecting the limit as batch size since it does not consider memory. However, I am unsure of how to pick a heuristic value.

Reply:
I suppose `batch_size` is the best fit out of all settings DF has at this moment -- there is no such setting (AFAIK) as `batch_size_bytes` / anything like adaptive batching based on memory size, so I suppose this logic to be more or less aligned with other operators.